### PR TITLE
feat: add amplitude to proposal list page and pagination

### PR DESCRIPTION
**Pull request by bkioshn**
## Describe your changes
Add amplitude track to proposal list page
- TO_PROPOSAL_LIST
- USE_INPUT (search)
- USE_FILTER_PROPOSALS_STATUS
- USE_FILTER_PROPOSALS_TYPE
- USE_FILTER_MY_PROPOSALS
- USE_CREATE_NEW_PROPOSAL
- USE_COPIER
proposal ID, block height, proposer
- USE_PAGINATION
left, right, page size, current page
- MINTSCAN
to page type, proposal type, proposal status

### CODE REVIEW

File: src/lib/pages/proposals/table/ProposalTableRow.tsx [hunk 1]
```diff
@@ -44,23 +45,29 @@ export const ProposalTableRow = ({
...
+          });
+          window.open(
+            `${getExplorerProposalUrl(
+              currentChainName
+            )}/${proposal.proposalId.toString()}`,
+            "_blank",
+            "noopener,noreferrer"
+          );
+        }
+      }}
```
Comment (poomthiti) [inline src/lib/pages/proposals/table/ProposalTableRow.tsx:62]: ternary

File: src/lib/services/amplitude.tsx [hunk 2]
```diff
@@ -174,8 +186,11 @@ export const AmpTrackUseRadio = (radio: string) =>
 export const AmpTrackUseOtherModal = (title: string) =>
   track(AmpEvent.USE_OTHER_MODAL, { title });
 
-export const AmpTrackMintscan = (type: string) =>
-  track(AmpEvent.MINTSCAN, { type });
+export const AmpTrackMintscan = (
+  type: string,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  properties?: Record<string, any>
```
Comment (poomthiti) [inline src/lib/services/amplitude.tsx:192]: Dict

File: src/lib/components/table/proposals/Proposer.tsx [hunk 3]
```diff
@@ -4,14 +4,21 @@ import { useGetAddressType } from "lib/app-provider";
...
+  amptrackSection?: string;
+}) => {
   const getAddressType = useGetAddressType();
 
   return proposer ? (
     <ExplorerLink
       type={getAddressType(proposer)}
       value={proposer}
       showCopyOnHover
+      ampCopierSection={amptrackSection}
```
Comment (poomthiti) [inline src/lib/components/table/proposals/Proposer.tsx:21]: just hardcode proposal list if its only used in that page

File: src/lib/components/table/proposals/ResolvedHeight.tsx [hunk 4]
```diff
@@ -21,6 +23,7 @@ export const ResolvedHeight = ({
       type="block_height"
       value={resolvedHeight.toString()}
       showCopyOnHover
+      ampCopierSection={amptrackSection}
```
Comment (poomthiti) [inline src/lib/components/table/proposals/ResolvedHeight.tsx:26]: just hardcode proposal list if its only used in that page

**Review by poomthiti — CHANGES_REQUESTED** [review]

File: src/lib/components/table/proposals/Proposer.tsx (hunk 3 above)
Comment (bkioshn) [reply src/lib/components/table/proposals/Proposer.tsx:21]: This proposer is also use in account details page

File: src/lib/components/table/proposals/ResolvedHeight.tsx (hunk 4 above)
Comment (bkioshn) [reply src/lib/components/table/proposals/ResolvedHeight.tsx:26]: also use in account details page

**Review by songwongtp — APPROVED** [review]

**Review by evilpeach — APPROVED** [review]

**Review by poomthiti — APPROVED** [review]

**Pull request merged**